### DISCUSSION TITLE
v4 refactor

### DISCUSSION THREAD

**Discussion by gyurielf** [gyurielf/svelte-tel-input#235] [General]
Hello there!

I've been working on the `v4` for a while now. 
(There was a bunch of real life thing happened to me what blocked this for weeks & months. Sorry but that's how life going.)
I refactor it almost from the ground, because I wanted to make it more straightforward and concise.

Among other things:
- It's using svelte v5. (runes)
- Improved country parsing, via using priorities.
- Fix cursor positioning problem
- Improve parsing performance
- Removes unnecessary reactivity

but here come the interesting part:
- It's drops the national parsing, only international format will be supported. (Of course, I can be convinced to keep it, but it seems the simplest and most logical.)
  Here is why:
  - Freakin' hard to parse and format (mask) and maintain this code.
  - Users have cached phone numbers in their browsers from different sites, these sites use regular inputs where users had to write his phone number entirely. So it makes the lives worse if the entered (and saved) formats are different from each other. This way both can be used with same number for the users, which is a great UX.

and what I'm scratching my head about right now:
[scrnli_16P5X09o5wPb3W.webm](https://github.com/user-attachments/assets/e74e19b1-3cfe-437c-a06f-dce286081b8a)

When a phone number input reaches its valid state(aka max length), should we prevent adding more numbers ?

If you have ideas guyz, let me know. :)

Thanks!

**Comment by rgon** [comment]
Sounds great. So what exactly is missing in this new version that we can contribute to?

**Comment by nosovk** [comment]
Would be nice to test v5 version, could you publish it behind @next tag in npm?

**Comment by gyurielf** [comment]
Here it is guyz.
svelte-tel-input@4.0.0
https://svelte-tel-input.vercel.app/

Sorry for the late answer and release. We bought a house and renovated it and had a baby in the meantime :)
